client/components/app-promo/index.jsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import assign from 'lodash/object/assign';
+import classNames from 'classnames';
+import noop from 'lodash/utility/noop';
+import Gridicon from 'components/gridicon';
+
+import analytics from 'analytics';
+import store from 'store';
+
+export default React.createClass( {
+
+	displayName: 'AppPromo',
+
+	getInitialState: function() {
+		var has_dismissed = store.get('desktop_promo_dismissed');
+		var promo_options = [
+			{ promo_code: 'a0001', message: 'WordPress.com your way  — apps now available in three delicious flavors — desktop, iOS, and Android.' },
+			{ promo_code: 'a0002', message: 'Get WordPress.com apps for desktop and mobile: download now!' },
+			{ promo_code: 'a0003', message: 'WordPress.com apps now available for desktop and mobile: download now!' },
+			{ promo_code: 'a0004', message: 'WordPress.com wherever you are — apps now available for desktop and mobile.' },
+			{ promo_code: 'a0005', message: 'WordPress.com at your fingertips — download apps for desktop and mobile.' },
+			{ promo_code: 'a0006', message: 'Blog anywhere, any time: download WordPress.com apps for desktop and mobile.' }
+		];
+
+		var item = promo_options[Math.floor(Math.random()*promo_options.length)];
+		return {
+			promo_item: item,
+			dismissed: has_dismissed
+		};
+	},
+
+	componentDidMount: function() {
+		// record promo view event
+		if (!this.state.dismissed ) {
+			analytics.tracks.recordEvent( 'calypso_desktop_promo_view', { 'promo_location': this.props.location, 'promo_code': this.state.promo_item.promo_code } );
+		}
+	},
+
+	recordClickEvent: function() {
+		analytics.tracks.recordEvent( 'calypso_desktop_promo_click', { 'promo_location': this.props.location, 'promo_code': this.state.promo_item.promo_code } );
+	},
+
+	dismiss: function() {
+		this.setState( { dismissed: true } );
+
+		// store as dismissed
+		store.set('desktop_promo_dismissed', true);
+	},
+
+	render() {
+		if ( this.state.dismissed ) {
+			return null;
+		}
+
+		var classes = classNames( 'app-promo', {
+			'dismissed': false
```

Comment:
if `dismissed` is always false, then is `classNames` and the `classes` variable even needed here?  As is `classes` will always be `app-promo`.